bootstrap/bootstrapper/fs/source.go

```diff
@@ -130,6 +120,7 @@ func (fss *fileSystemSource) ReadData(shard uint32, tr xtime.Ranges) (m3db.Shard
 			block.Reset(timeRange.Start, encoder)
 			curMap.AddBlock(id, block)
 		}
+		hasError = hasError || r.Validate() != nil
```

Comment:
Perhaps log the `Validate()` error if it fails so we can see why bootstrap is failing.